main.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    counter = 0
+    while counter < size
+      yield(self[counter])
+      counter += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    output = []
+    my_each { |ele| output << ele if yield(ele) }
+    output
+  end
+
+  def my_all?
+    output = true
+    my_each do |ele|
+      output = false if yield(ele) != true
+    end
+    output
+  end
+
+  def my_any?
+    output = false
+    my_each do |ele|
+      output = true if yield(ele) == true
+    end
+    output
+  end
+
+  def my_none?
+    output = true
+    my_each do |ele|
+      output = false if yield(ele) == true
+    end
+    puts output
+  end
+
+  def my_count(number = nil)
+    count = 0
+    return size if number.nil? && !block_given?
```

Comment:
:thinking: Is it possible to pass a number as an argument and also pass a block for `count` or `my_count` methods?

```ruby
$ irb
irb(main):001:0> [1,2,3,4,5].count(3){|i| i < 5}
(irb):1: warning: given block not used
=> 1
```
Can you give a use case for this, beacuse I don't think we need to have both `arg` and `block` for `#my_count` method? :thinking: 
See the documentation on ruby's [`count`](https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-count) method.